### PR TITLE
Use cross-platform line separator

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java [hunk 1]
```diff
@@ -176,7 +176,6 @@ private static String positionalConcat(String line, Position startPos, Position
 
     private static List<String> split(String text)
     {
-        return Stream.of(splitByWholeSeparatorPreserveAllTokens(text, System.lineSeparator()))
-                     .collect(Collectors.toList());
+        return Splitter.on(LINE_SEPARATOR_PATTERN).splitToStream(text).collect(Collectors.toList());
```
Comment (valfirst) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java:179]: why not `String#split`?

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java [hunk 2]
```diff
@@ -45,6 +45,8 @@
 public class TextDocumentEditor implements TextDocumentEventListener, TextDocumentProvider
 {
     private static final Logger LOGGER = LoggerFactory.getLogger(TextDocumentEditor.class);
+    private static final Pattern LINE_SEPARATOR_PATTERN = Pattern.compile("\\R");
```
Comment (valfirst) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java:48]: how will it match `\r\n`?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java (hunk 1 above)
Comment (uarlouski) [reply vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java:179]: it was initially done with splitByWholeSeparatorPreserveAllTokens to preserve all tokens, just splilt does not preserve tokens

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java (hunk 2 above)
Comment (uarlouski) [reply vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java:48]: Pattern.compile("\\R").matcher("\r\n").matches() -> true

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java (hunk 1 above)
Comment (valfirst) [reply vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java:179]: `java.lang.String#split(java.lang.String, int)` with limit = -1 ?

Comment (uarlouski) [reply vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java]: fixed for now https://github.com/vividus-framework/vividus-studio/issues/327 :)

**Review by ikalinin1 — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**